packages/vite/src/node/plugins/resolve.ts

```diff
@@ -531,6 +525,9 @@ function tryFsResolve(
   if (res) return res + postfix
 }
 
+const knownTsOutputRE = /\.(?:js|mjs|cjs|jsx)$/
+const isPossibleTsOutput = (url: string): boolean => knownTsOutputRE.test(url)
```

Comment:
Moved this function here because it isn't a widely used util and it only works without query, that is what we need in `tryCleanFsResolve`. I think we are pushing too many functions to `utils` lately and we will be better of delaying moving them there to avoid creating abstractions that aren't really needed